### ISSUE TITLE
False positive: Value stored is never read

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#18322]
|  |  |
| --- | --- |
| Bugzilla Link | [17948](https://llvm.org/bz17948) |
| Version | trunk |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 
This code triggers a warning:
> clang-problems.cpp:, Clang (LLVM based), Priority: Normal
> Value stored to 'i' is never read. 
Value is read in another thread.

```c++
struct Test
{
public:
    Test(bool& i_bool)
        : the_bool(i_bool)
    {}

    void
    run()
    {
        while(the_bool);
    }
    bool& the_bool;
};

void
schedule_test(Test* t)
{
    // puts it on a queue that is handled in another thread
    (void)t;
}

void f()
{
    bool i = false;
    Test* t = new Test(i);
    schedule_test(t);
    i = true;
    // wait for threadpool to become empty
}
```

**Comment by philnik777** [comment]
The diagnostic is correct. This is not a valid way to synchronize threads.
